app/components/CrashDiagramCard.tsx

```diff
@@ -2,63 +2,166 @@ import { useState } from "react";
+    </div>
+  );
+};
 export default function CrashDiagramCard({ crash }: { crash: Crash }) {
   const [diagramError, setDiagramError] = useState(false);
+  const [rotation, setRotation] = useState(0);
 
   return (
     <Card>
       <Card.Header>Diagram</Card.Header>
       <Card.Body className="crash-header-card-body text-center">
-        {!diagramError && (
-          <Image
-            fluid
-            style={{
-              height: "100%",
-              maxWidth: "100%",
-            }}
-            src={`${CR3_DIAGRAM_BASE_URL}/${crash.record_locator}.jpeg`}
-            alt="crash diagram"
-            onError={() => {
-              console.error("Error loading CR3 diagram image");
-              setDiagramError(true);
-            }}
-          />
-        )}
-        {diagramError && crash.is_temp_record && (
-          <Alert variant="info">
-            <i className="fa fa-info-circle" /> Crash diagrams are not available
-            for user-created crash records
-          </Alert>
-        )}
-        {diagramError && !crash.is_temp_record && (
-          <Alert variant="danger">
-            <p>
-              The crash diagram is not available. Typically, this indicates
-              there was an error when processing this crash&aposs CR3 PDF.
-            </p>
-            <p>
-              For additional assistance, you can&nbsp;
-              <a
-                href="https://atd.knack.com/dts#new-service-request/?view_249_vars=%7B%22field_398%22%3A%22Bug%20Report%20%E2%80%94%20Something%20is%20not%20working%22%2C%22field_399%22%3A%22Vision%20Zero%20Editor%22%7D"
-                target="_blank"
-                rel="noopener noreferrer"
-              >
-                report a bug
-                <i className="fa fa-external-link"></i>
-              </a>
-              .
-            </p>
-          </Alert>
-        )}
+        <TransformWrapper>
+          {!diagramError && <ZoomResetControls setRotation={setRotation} />}
+          <TransformComponent>
+            <div style={{ height: "100%", overflow: "hidden" }}>
+              {!diagramError && (
+                <Image
+                  fluid
+                  style={{
+                    height: "100%",
+                    width: "100%",
+                    transform: `rotate(${rotation}deg)`,
+                    objectFit: "contain",
+                  }}
+                  src={`${CR3_DIAGRAM_BASE_URL}/${crash.record_locator}.jpeg`}
+                  alt="crash diagram"
+                  onError={() => {
+                    console.error("Error loading CR3 diagram image");
+                    setDiagramError(true);
+                  }}
+                />
+              )}
+              {diagramError && crash.is_temp_record && (
+                <Alert variant="info" style={{ marginTop: "50px" }}>
+                  <i className="fa fa-info-circle" />
+                  Crash diagrams are not available for temporary crash records
```

Comment:
I changed this language. It felt accusatory that this wouldn't work because it was "user-made." This feels more like a statement of fact using commonly understood terms.

Reply:
Could there be the case though that the crash diagram is not related for a non-temporary crash? Like if there is a processing error or some other error?

Reply:
ah now I scrolled down and see that you included a different message. I should scroll more before i speak

Reply:
I like this change—thank you!